guardian/support-service-lambdas
# AWS Api User Credentials Setter

## Feature Description
Generate and set a password for each Aws API User in Salesforce and then update a Secret in Secrets Manager with the new password value

## Context
This feature has been developed to relieve some of the manual effort involved in API User Activation in Salesforce when a Sandbox Refresh has taken place.

## Process Flow
![image](https://user-images.githubusercontent.com/36296660/96882507-48cd2600-1477-11eb-9e97-e1d483bd9aad.png)

## Testing

1. Identify a test user in the sandbox
2. Alter the query in the lambda to query for that user only
3. Execute the lambda
4. navigate to secrets manager
5. Verify that an entry has been created for the user OR that an existing entry has been updated
6. Use the password stored in the secrets manager to log in as the user
7. If log in was successful, then the process has executed successfully

## Considerations

- The running user for this process will be generated at the point of sandbox generation, therefore at the point of deployment, there will be no credentials available

- Salesforce Password Policy is currently:
a minimum of 8 characters AND
at least 1 alpha character AND
at least 1 numeric character
	
An improvement to the solution would be if we could extract a regex representation from Salesforce programmatically for the Password Policy and inject this into the password generation mechanism
